### PR TITLE
refactor: application configuration

### PR DESCRIPTION
**Pull request by daniel-mader**
# Description of change
Refactor of UniCore's configuration.

Introduces one single configuration file. All configuration values can be overwritten using environment variables.

## Links to any relevant issues
n/a

## How the change has been tested
Manually tested using UniMe.

## Definition of Done checklist
Add an `x` to the boxes that are relevant to your changes.

- [x] I have followed the contribution guidelines for this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have successfully tested this change in a docker environment

### CODE REVIEW

File: agent_shared/src/config.rs [hunk 1]
```diff
@@ -1,52 +1,304 @@
...
+        let config = if cfg!(feature = "test_utils") {
+            config::Config::builder()
+                .add_source(config::File::with_name("../agent_shared/tests/test-config.yaml"))
+                // TODO: other prefix for tests
+                .add_source(config::Environment::with_prefix("TEST_AGENT").separator("__"))
+                .build()?
+        } else {
+            config::Config::builder()
+                .add_source(config::File::with_name("agent_application/example-config.yaml"))
+                .add_source(config::Environment::with_prefix("AGENT").separator("__"))
```
Comment (nanderstabel) [inline agent_shared/src/config.rs]: Should we start using `UNICORE` as a prefix?

File: agent_application/CONFIGURATION.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+| `CORS_ENABLED`                        | Enable CORS (permissive, allow all). Only required for web-based wallets. | `false`       | boolean                                                            |
+| `DID_METHOD_WEB_ENABLED`              | Create and host a `did:web` document.                                     | `false`       | boolean                                                            |
+| `DOMAIN_LINKAGE_ENABLED`              | Enable domain linkage (only works with `did:web`).                        | `false`       | boolean                                                            |
+| `EXTERNAL_SERVER_RESPONSE_TIMEOUT_MS` | The timeout for external server responses.                                | `1000`        | integer                                                            |
+| `PREFERRED_DID_METHOD`                | The default DID method to use.                                            | `jwk`         | `jwk`, `key`, `web`                                                |
```
Comment (nanderstabel) [inline agent_application/CONFIGURATION.md:20]: Shouldn't all the variable names be prefixed or is that deliberately omitted?

File: agent_store/src/postgres.rs [hunk 3]
```diff
@@ -67,8 +67,8 @@ where
     }
 }
 
-pub async fn issuance_state(event_publishers: Vec<Box<dyn EventPublisher>>) -> IssuanceState {
-    let pool = default_postgress_pool(&config!("db_connection_string", String).unwrap()).await;
+pub async fn issuance_state(event_publishers: Vec<Box<dyn EventPublisher>>, connection_string: &str) -> IssuanceState {
```
Comment (nanderstabel) [inline agent_store/src/postgres.rs]: Why do we add the connection_string to the function's signature here? Can't we just directly retrieve it from the `config()`?

Comment (daniel-mader) [reply agent_store/src/postgres.rs]: Refactored ✅

**Review by nanderstabel — DISMISSED** [review]

**Review by nanderstabel — APPROVED** [review]

**Pull request merged**